scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -127,11 +130,15 @@ getLedgerData() { # getNodeMetrics expected to have been already run
 }
 
 getConsensus() {
-  getProtocolParams
+  if isNumber "$1"; then
+     getProtocolParamsHist "$(( $1 - 1 ))" || return 1
+  else
+     getProtocolParams || return 1
+  fi
   if versionCheck "9.0" "${PROT_VERSION}"; then
     consensus="cpraos"
     stability_window_factor=3
-  elif versionCheck "8.0" "${PROT_VERSION}"; then
+  elif versionCheck "7.0" "${PROT_VERSION}"; then
```

Comment:
was this intentional?

Reply:
Hello @rdlrt 

the condition I found in getConsensus function of cncli.sh when I started the work was:

>getConsensus() {
  getProtocolParams
  if versionCheck "10.0" "${PROT_VERSION}"; then
    consensus="cpraos"
    stability_window_factor=3
  elif versionCheck "7.0" "${PROT_VERSION}"; then
    consensus="praos"
    stability_window_factor=2
  else
    consensus="tpraos"
    stability_window_factor=2
  fi
}

I've just completed the upgrade to node 9.1 using latest guild-deploy / CNTools 

and condition now is 

>getConsensus() {
  getProtocolParams
  if versionCheck "9.0" "${PROT_VERSION}"; then
    consensus="cpraos"
    stability_window_factor=3
  elif versionCheck "8.0" "${PROT_VERSION}"; then
    consensus="praos"
    stability_window_factor=2
  else
    consensus="tpraos"
    stability_window_factor=2
  fi
}

I also just checked in the backup files in archive dir from the upgrade.
And noticed since days that in alpha was as the latest consensus condition.

Cheers,
Manuel

Reply:
Yeah this error snuck in there for a short time by mistake. Just bad luck that you managed to base your code with that error.  

8.0 I'm fairly certain is the correct major version for this.

Reply:
Hello @Scitz0 and @rdlrt 

I updated getConsensus as

>getConsensus() {
  if isNumber "$1" ; then
     getProtocolParamsHist "$(( $1 - 1 ))" || return 1
  else
     getProtocolParams || return 1
  fi

  >if versionCheck "9.0" "${PROT_VERSION}"; then
    consensus="cpraos"
    stability_window_factor=3
  elif versionCheck "8.0" "${PROT_VERSION}"; then
    consensus="praos"
    stability_window_factor=2
  else
    consensus="tpraos"
    stability_window_factor=2
  fi
}

and started to test the epochs I have with set -x, in this way consensus praos is used from epoch 395, I had a schedule on 394, last epoch using tpraos. The change from 7.0 to 8.0 makes use tpraos for 30 epochs more.

I remember that praos consensus starts in epoch 364, I can be wrong, I'll run leaderlog in both ways for some previous epoch to check for different calculation.

Reply:
Hello @Scitz0

I tested about consensus and protocol version:

this returns correct slots calculation

elif versionCheck "7.0" "${PROT_VERSION}"; then
consensus="praos"

>epoch,epoch_nonce,sigma,epoch_slots_ideal,max_performance
380,3950677ef48bb212ad335f253ad1998be4e9fe4ae2a26d402607850461398125,0.0000015520185588821839,0.03,3333.33
387,6d49fffa700bb7d3399e2dcad064feb9d881900e80e3148471bd5029382ac4da,0.0000016038306633996149,0.03,6666.67

using 8.0 calculation is wrong

elif versionCheck "8.0" "${PROT_VERSION}"; then
consensus="praos"

>epoch,epoch_nonce,sigma,epoch_slots_ideal,max_performance
>380,3950677ef48bb212ad335f253ad1998be4e9fe4ae2a26d402607850461398125,0.0000015520185588821839,0.03,0.0
387,6d49fffa700bb7d3399e2dcad064feb9d881900e80e3148471bd5029382ac4da,0.0000016038306633996149,0.03,0.0

a manual check at epoch/version change

>curl -s "https://api.koios.rest/api/v1/epoch_params?_epoch_no=364" | jq -r '.[].protocol_major,.[].protocol_minor'
6
0

>curl -s "https://api.koios.rest/api/v1/epoch_params?_epoch_no=365" | jq -r '.[].protocol_major,.[].protocol_minor'
7
0

>curl -s "https://api.koios.rest/api/v1/epoch_params?_epoch_no=393" | jq -r '.[].protocol_major,.[].protocol_minor'
7
0

>curl -s "https://api.koios.rest/api/v1/epoch_params?_epoch_no=394" | jq -r '.[].protocol_major,.[].protocol_minor'
8
0

cheers
Manuel

Reply:
Ok thanks for the tests. Then let's switch to 7.0 👍🏻

Reply:
Hello @Scitz0 

thanks for confirming

cheers,
Manuel